lib/proto.js

```diff
@@ -85,6 +86,19 @@ function child (bindings) {
   return instance
 }
 
+function childBindings () {
+  const chindings = this[chindingsSym]
+  var chindingsObject = (chindings.indexOf(',') === 0) ? `{${chindings.substr(1)}}` : `{${chindings}}`
+  try {
+    var bindings = JSON.parse(chindingsObject) || {}
+    delete bindings['pid']
+    delete bindings['hostname']
```

Comment:
I wonder can there be slice out of the string before it gets parsed, this will mean that JSON.parse has less to process, and will return an object that has been deoptimized out of fast properties (https://v8.dev/blog/fast-properties) – `delete` is bad for objects

Reply:
yes, I also thought about this, but I have no idea how the string looks like. My current examples (via tests) are 
```js
// ,"pid":1180,"hostname":"myYoga"
// or
// ,"pid":1180,"hostname":"myYoga","name":"basicTest","foo":"bar","a":2
```
If I could be sure that pid and hostname are always near each other, in this order, than I could just strip them away with a regex. But that could be dangerous and if I make sure that nothing else will be removed the code will be of a slow performance.

I have read about delete being bad an the alternative is to set the properties to undefined. What do you think about that?

Reply:
I think `delete`  is fine here, let's leave optimizations to the next steps. This should not sit in a hot path, and in case we'll have a cache and a fast properties conversion in place.